### PR TITLE
Write version in pyproject toml if available

### PR DESCRIPTION
**Pull request by mauritsvanrees**
- Added `pyproject.toml` and moved most metadata there. "Eat your own dog food." I am not sure if we can get rid of the remaining parts of `setup.py`.
- Add reader. I added `tomli` to the dependencies on Pythons earlier than 3.11. Chances are that you already have this via a different dependency. On 3.11 I use the built-in `tomllib`. 
- Add writer. This is not that different from how we write to `setup.py` or other files.
- Checked that `bin/prerelease` updates the version in our own `pyproject.toml`. I did not commit this.

This fixes issue #373.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,2 +1,91 @@
...
+    "Intended Audience :: Developers",
+    "License :: OSI Approved :: GNU General Public License (GPL)",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.6",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
```
Comment (reinout) [inline pyproject.toml:29]: Add 3.11 too?

File: setup.py [hunk 2]
```diff
@@ -26,88 +23,9 @@ def read(filename):
...
-        "Programming Language :: Python :: Implementation :: CPython",
-        "Programming Language :: Python :: Implementation :: PyPy",
-        "Topic :: Software Development :: Libraries :: Python Modules",
-    ],
-    keywords=["releasing", "packaging", "pypi"],
-    author="Reinout van Rees",
-    author_email="reinout@vanrees.org",
-    url="https://zestreleaser.readthedocs.io",
-    license="GPL",
     packages=find_packages(exclude=["ez_setup"]),
```
Comment (reinout) [inline setup.py:27]: We can probably zap `ez_setup` altogether? Wasn't that sort of a setuptools bootstrap file for when setuptools was new? I don't totally remember :-)

File: zest/releaser/vcs.py [hunk 3]
```diff
@@ -233,6 +249,50 @@ def _update_python_file_version(self, version):
         utils.write_text_file(filename, contents, encoding)
         logger.info("Set __version__ in %s to '%s'", filename, version)
 
+    def _update_pyproject_toml_version(self, version):
```
Comment (reinout) [inline zest/releaser/vcs.py:252]: Test?

**Review by reinout — COMMENTED** [review]
Good idea!
Most important point: I think a test has to be added? There will be corner cases later on, I fear.

File: pyproject.toml (hunk 1 above)
Comment (mauritsvanrees) [reply pyproject.toml:29]: Done.

File: setup.py (hunk 2 above)
Comment (mauritsvanrees) [reply setup.py:27]: Right, it only makes sense to exclude `ez_setup.py` if we actually have this file, which we don't. I removed the code.

File: zest/releaser/vcs.py (hunk 3 above)
Comment (mauritsvanrees) [reply zest/releaser/vcs.py:252]: Done. See commit 1bd131523698c383d6974ded8c55a95355b498c6.

**Comment by mauritsvanrees** [comment]
Okay, so I added a test, and it works fine. The message from the commit may be interesting: 1bd131523698c383d6974ded8c55a95355b498c6.

Problem: the test does not work on Python 3.6, so I dropped support for that. We will want to make a major release for this.
And the test shows that if you *only* have a `pyproject.toml` and no `setup.py` at all, then `fullrelease` does not create and upload distributions to PyPI. So that is something that we need to fix. That is basically issue #295.

I think the current PR is fine to merge: it enables updating the version when it is in `pyproject.toml`. But releasing still needs a `setup.py` at this point, though it can be mostly empty.
We could make an 8.0.0a1 release with this.

**Comment by reinout** [comment]
I agree. I'll merge it.
